web/src/pages/playground.tsx

```diff
@@ -0,0 +1,253 @@
+            "using the following sources. Each source is labeled with a letter. Feel free to " +
+            "use the sources in any order, and try to use multiple sources in your answers.\n\n"
+        ),
+        suffix: (
+            "\n\n" +
+            "Before the question (\"Q: \"), there will be a history of previous questions and answers. " +
+            "These sources only apply to the last question. Any sources used in previous answers " +
+            "are invalid."
+        ),
+    },
+    question: (
+        "In your answer, please cite any claims you make back to each source " +
+        "using the format: [a], [b], etc. If you use multiple sources to make a claim " +
+        "cite all of them. For example: \"AGI is concerning [c, d, e].\"\n\n"
+    ),
+    modes: {
+        default: "",
+        concise: (
+            "Answer very concisely, getting to the crux of the matter in as " +
+            "few words as possible. Limit your answer to 1-2 sentences.\n\n"
+        ),
+        rookie: (
+            "This user is new to the field of AI Alignment and Safety - don't " +
+            "assume they know any technical terms or jargon. Still give a complete answer " +
+            "without patronizing the user, but take any extra time needed to " +
+            "explain new concepts or to illustrate your answer with examples. " +
+            "Put extra effort into explaining the intuition behind concepts " +
+            "rather than just giving a formal definition.\n\n"
+        ),
+    }
+}
+const DEFAULT_SETTINGS = {
+    prompts: DEFAULT_PROMPTS,
+    mode: 'default',
+    completions: 'gpt-3.5-turbo',
+    encoder: 'cl100k_base',
+    topKBlocks: 10,         //  the number of blocks to use as citations
+    numTokens: 4095,
+    tokensBuffer: 50,      //  the number of tokens to leave as a buffer when calculating remaining tokens
+    maxHistory: 10,        //  the max number of previous items to use as history
+    historyFraction: 0.25, //  the (approximate) fraction of num_tokens to use for history text before truncating
+    contextFraction: 0.5,  //  the (approximate) fraction of num_tokens to use for context text before truncating
+}
+const COMPLETION_MODELS = ['gpt-3.5-turbo', 'gpt-4']
+const ENCODERS = ['cl100k_base']
+
+const updateIn = (obj, [head, ...rest], val) => {
+    if (!head) {
+        // No path provided - do nothing
+    } else if (!rest ||rest.length == 0) {
+        obj[head] = val;
+    } else {
+        updateIn(obj[head], rest, val)
+    }
+    return obj
+}
+
+type ChatSettingsParams = {
+    settings: LLMSettings,
+    updateSettings: (updater: (settings: LLMSettings) => LLMSettings) => void
+}
+type InputFields = {
+    field: string,
+    label: string,
+    min?: number,
+    max?: number,
+    parser?: (v: string) => number,
+}
+
+const ChatSettings = ({settings, updateSettings}: ChatSettingsParams) => {
```

Comment:
![image](https://github.com/StampyAI/stampy-chat/assets/3942390/904a99f2-9336-4e4d-8698-24f88651e5be)

Basically a list of possible numerical values